scripts/switch-yarn-to-pnpm.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+set -euxo pipefail
+
+# Helper functions and setup of jq
+file_exists() {
+  [ -f "$1" ]
+}
+
+command_exists() {
+  command -v "$1" >/dev/null 2>&1
+}
+
+if ! command_exists jq; then
```

Comment:
Feel free to ignore but... I wonder if we should just bite the bullet and write this script in Ruby? I made the mistake of writing the original ones in bash cos I thought they would be simple... but complexity has ballooned over time 😢

Reply:
> in Ruby

some time ago I started redoing existing scripts in node js 😅 we could revive that effort, but it's low priority. though we could at least land the support for js scripts (w/o actually finishing the rewrite of all of the existing stuff)

Reply:
> land the support for js scripts

It should 'just work', right? As long as you change the shebang at the top of the file to something like

```
#!/usr/bin/env node
```
or for ruby

```
#!/usr/bin/env ruby
```
?

Reply:
Oh, yeah, I suppose that would work too. I was thinking about "native" support, without shelling out, but I guess that would make more sense when all scripts are already in js.